docs/proposals/20250724-independent-worker-topology-versions.md

```diff
@@ -0,0 +1,515 @@
+	/*
+    Existing fields omitted
+  */
+
+	// Version specifies the Kubernetes version for this MachineDeployment worker node group.
+	// When set, this version overrides the cluster-wide version specified in Cluster.Spec.Topology.Version
+	// for this specific MachineDeployment, enabling independent version management and upgrade scheduling.
+	//
+	// Version skew constraints as defined by Kubernetes' Version Skew Policy are enforced between
+	// this worker node group version and the control plane version. The worker nodes cannot be more
+	// than 2 minor versions behind the control plane (e.g., control plane at v1.30.x supports
+	// worker nodes down to v1.28.x).
+	//
+	// This field enables use cases such as:
+	// - Staggered upgrades across different worker node groups
+	// - Testing upgrades on a subset of worker nodes before rolling out cluster-wide
+	// - Managing different upgrade cadences for workloads with varying availability requirements
+	//
+	// If not specified, the MachineDeployment will use the version from Cluster.Spec.Topology.Version.
+	// +optional
+	// +featureGate=IndependentWorkerTopologyVersion
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Version *string `json:"version,omitempty"`
+}
+```
+- We will also initially put this functionality behind a feature flag. 
+  - We will follow the same approach that the ClusterTopology flag takes. So, for example, if we create a cluster with the flag enabled, and then remove the flag from the controller, the controller webhooks would reject any version changes to the created cluster both for `topology.version` and `machinedeployment/machinepool.version`. 
+
+- What are some important details that didn't come across above.
+- What are the caveats to the implementation?
+- Go in to as much detail as necessary here.
+- Talk about core concepts and how they relate.
+
+### Security Model
+
+Document the intended security model for the proposal, including implications
+on the Kubernetes RBAC model. Questions you may want to answer include:
+
+* Does this proposal implement security controls or require the need to do so?
+  * If so, consider describing the different roles and permissions with tables.
+* Are their adequate security warnings where appropriate (see https://shostack.org/files/papers/ReederEtAl_NEATatMicrosoft.pdf for guidance).
+* Are regex expressions going to be used, and are their appropriate defenses against DOS.
+* Is any sensitive data being stored in a secret, and only exists for as long as necessary?
+
+### Risks and Mitigations
+
+- What are the risks of this proposal and how do we mitigate? Think broadly.
+- How will UX be reviewed and by whom?
+- How will security be reviewed and by whom?
+- Consider including folks that also work outside the SIG or subproject.
+
+## Alternatives
+
+The `Alternatives` section is used to highlight and record other possible approaches to delivering the value proposed by a proposal.
+
```

Comment:
@ivelichkovich seems like we've identified two approaches: 
1. Don't use ClusterClass altogether.
2. Manage `MachineDeployments` outside of the `ClusterClass` and point them to the Cluster